File: 370-range-addition/370-range-addition.py
class Solution:
    def getModifiedArray(self, length: int, updates: List[List[int]]) -> List[int]:
        res = [0] * length
        for update in updates:
            value = update[2]
            start = update[0]
            end = update[1]
            res[start] += value
            
            if end + 1 < length:
                res[end + 1] -= value
        arrSum = 0
        for i in range(len(res)):
            arrSum += res[i]
            res[i] = arrSum
        return res

File: 370-range-addition/370-range-addition.java
class Solution {
    public int[] getModifiedArray(int length, int[][] updates) {
        int[] ans = new int[length];
        for (int i = 0; i < updates.length; i++) {
            int startIndx = updates[i][0];
            int endIndx = updates[i][1];
            int val = updates[i][2];
            ans[startIndx] += val;
            if (endIndx + 1 < length) ans[endIndx + 1] -= val;
        }
        
        for (int i = 1; i < length; i++) {
            ans[i] += ans[i - 1];
        }
        return ans;
    }
}